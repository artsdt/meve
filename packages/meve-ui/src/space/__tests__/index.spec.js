import Space from '..'
import { mount } from '@vue/test-utils'

test('test space basic use', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template: '<m-space>' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test internal size ', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space size="large">' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test custom size ', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space :size="[50, 50]">' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space align center', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space align="center">' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space justify center', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space justify="center">' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space justify around', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space justify="space-around">' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space justify between', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space justify="space-between">' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space wrap', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space :wrap="false">' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space direction', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template:
      '<m-space direction="column">' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '<span>hello</span>' +
      '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test space inline', () => {
  const wrapper = mount({
    components: {
      [Space.name]: Space,
    },
    template: '<m-space inline>' + '<span>hello</span>' + '<span>hello</span>' + '<span>hello</span>' + '</m-space>',
  })

  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
