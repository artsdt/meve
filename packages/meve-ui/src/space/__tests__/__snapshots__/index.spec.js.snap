// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test custom size  1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: start; flex-wrap: wrap; margin: -25px 0px;\\">
  <div style=\\"margin: 25px 50px 25px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 25px 50px 25px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 25px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test internal size  1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: start; flex-wrap: wrap; margin: -6px 0px;\\">
  <div style=\\"margin: 6px 20px 6px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 6px 20px 6px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 6px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space align center 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: start; align-items: center; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space basic use 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: start; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space direction 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: column; justify-content: start; flex-wrap: wrap;\\">
  <div style=\\"margin: 0px 0px 8px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 0px 0px 8px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space inline 1`] = `
"<div class=\\"m-space m--box m--inline-flex\\" style=\\"flex-direction: row; justify-content: start; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space justify around 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: space-around; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 6px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 6px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 6px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space justify between 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: space-between; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 6px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 6px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px 4px 6px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space justify center 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: center; flex-wrap: wrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px;\\"><span>hello</span></div>
</div>"
`;

exports[`test space wrap 1`] = `
"<div class=\\"m-space m--box\\" style=\\"flex-direction: row; justify-content: start; flex-wrap: nowrap; margin: -4px 0px;\\">
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 12px 4px 0px;\\"><span>hello</span></div>
  <div style=\\"margin: 4px 0px;\\"><span>hello</span></div>
</div>"
`;
